modules/core/include/opencv2/core/hal/intrin_cpp.hpp

```diff
@@ -828,6 +829,24 @@ template<typename _Tp, int n> inline v_reg<typename V_TypeTraits<_Tp>::w_type, n
     return c;
 }
 
+/** @brief Dot product and add
+
+Same as cv::v_dotprod, but add a third element to the sum of adjacent pairs.
+
+Scheme:
+@code
+  {A1 A2 ...} // 16-bit
+x {B1 B2 ...} // 16-bit
+-------------
+  {A1B1+A2B2+C1 ...} // 32-bit
+
+@endcode
+Implemented only for 16-bit signed source type (v_int16x8).
+*/
+
+inline v_int32x4 v_dotprod_add(const v_int16x8& a, const v_int16x8& b, const v_int32x4& c)
```

Comment:
Perhaps we can reuse `v_dotprod()` name with additonal parameter for initial accumulator value.

/cc @mshabunin ?

Reply:
So this part was bit suspicious.
If we combine ```v_dotprod``` and ```v_dotprod_add``` there might be unnecessary "Plus Zero" operation every where in the source code.
I did some experiments, and at least Visual Studio seems to be good enough to wipe out "Plus Zero" instruction.
I hope GCC is good enough too, so I reverted ```v_dotprod_add``` and combined with the existing ```v_dotprod```.

Reply:
Oops, I've missed this comment.
Let me put back the ```v_dotprod```